### PR TITLE
refactor: chunk InfluxRPC `ReadResponses`

### PR DESCRIPTION
**Pull request by crepererum**
Currently we see some prod panics:

```
'assertion failed: len <= std::u32::MAX as usize', tonic/src/codec/encode.rs:127:5
```

This is due to an upstream bug in tonic:
https://github.com/hyperium/tonic/issues/1141

However the fix will only turn this into an error instead of panicking. We should instead NOT return such overlarge results, esp. because InfluxRPC supports streaming.

While we currently don't perform streaming conversion (like streaming the data out of the query stack into the gRPC layer), the 4GB size limit can easily be triggered (in prod) w/ enough RAM. So let's re-chunk our in-memory responses so that they stream nicely to the client.

We may later implement proper streaming conversion, see #4445 and #503.

### CODE REVIEW

File: service_grpc_influxrpc/src/service.rs [hunk 1]
```diff
@@ -1586,6 +1595,58 @@ where
...
+
+    let mut frames = vec![];
+    let mut size = 0;
+    for frame in it {
+        let fsize = frame_size(&frame);
+
+        // flush?
+        if size + fsize > size_limit {
+            let mut tmp = vec![];
+            std::mem::swap(&mut tmp, &mut frames);
```
Comment (alamb) [inline service_grpc_influxrpc/src/service.rs:1619]: I think you can write this pattern using `take`, which may be slightly easier to read

```suggestion
            let frames =  std::mem::take(&mut frames);
```

**Review by alamb — APPROVED** [review]
Looks good to me -- writing some tests might be good

File: service_grpc_influxrpc/src/service.rs (hunk 1 above)
Comment (crepererum) [reply service_grpc_influxrpc/src/service.rs]: done

**Comment by crepererum** [comment]
> writing some tests might be good

Added unit tests for the chunking routine.

**Review by alamb — COMMENTED** [review]
Thanks for the tests!

**Pull request merged**